### PR TITLE
Fix codeql errors

### PR DESCRIPTION
**Pull request by pskiran1**
This pull request aims to address various CodeQL errors identified in the codebase:

- Removal of unused variables and imports
- Elimination of statements with no effect
- Resolving issues related to multiple definitions with the same function names
- Correcting incorrect operator usage, assert types in test scripts, and ensuring proper parent class constructor calling in subclasses
- Avoiding overwriting attributes in the super-class or sub-class that have already been initialized.
- Addressing problems with undefined methods, variables, improper exception handling, and commented-out code
- Rectification of explicit and implicit return type combinations, including conditional returns and missing return data types.

### CODE REVIEW

File: tests/test_search_dimensions.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def test_list_behavior(self):
         self.assertEqual(count, 3)
 
         with self.assertRaises(IndexError):
-            dim3 = dims[3]
```
Comment (nv-braf) [inline tests/test_search_dimensions.py:64]: I understand the change gets rid of the codeQL error, but it still seems to me like the line isn't doing anything

File: model_analyzer/config/input/config_command.py [hunk 2]
```diff
@@ -36,7 +36,6 @@ def __init__(self):
         """
 
         self._fields = {}
-        self._fill_config()
```
Comment (tgerdesnv) [inline model_analyzer/config/input/config_command.py:39]: Are you sure about this change? Does our CI pass?  I know that ConfigCommand hasn't implemented it, but I think there are child classes that rely on this function call.

Comment (tgerdesnv) [reply model_analyzer/config/input/config_command.py:39]: I see now that you manually added the function call to all the child classes, and that the recommendation is to not call subclass functions in init.  Probably better ways to handle this with interface class or calling fill_config outside of init, but I'm fine to leave this how you have it.

File: tests/test_perf_analyzer_config_generator.py [hunk 3]
```diff
@@ -712,7 +712,7 @@ def _test_throughput_gain_valid_helper(self, throughput_values, expected_result)
             for throughput_value in throughput_values
         ]
 
-        pacg = PerfAnalyzerConfigGenerator(
+        PerfAnalyzerConfigGenerator(
```
Comment (tgerdesnv) [inline tests/test_perf_analyzer_config_generator.py:715]: Just delete this whole line. We aren't using the created PACG

File: tests/test_search_dimensions.py (hunk 1 above)
Comment (tgerdesnv) [reply tests/test_search_dimensions.py:64]: The line is reading an out-of-range index, which should raise an IndexError assert. This change looks correct to me.

**Review by tgerdesnv — COMMENTED** [review]
This all looks great! Thanks for doing this

**Review by tgerdesnv — APPROVED** [review]
LGTM if CI is passing

File: model_analyzer/perf_analyzer/perf_config.py [hunk 4]
```diff
@@ -295,6 +295,7 @@ def remove_url_from_cli_string(cls, cli_string):
             perf_str_tokens.pop(url_index)
             perf_str_tokens.pop(url_index)
         except ValueError:
+            # Ignore ValueError exception
```
Comment (nv-hwoo) [inline model_analyzer/perf_analyzer/perf_config.py:298]: Unnecessary comment? (unsure if this is needed for codeql fix)

File: model_analyzer/perf_analyzer/perf_analyzer.py [hunk 5]
```diff
@@ -349,7 +348,8 @@ def _get_process_output(self):
         result = ""
         try:
             result = tmp_output.decode("utf-8")
-        except:
+        except Exception:
+            # Ignore the result on decode failed
```
Comment (nv-hwoo) [inline model_analyzer/perf_analyzer/perf_analyzer.py:352]: Unnecessary comment?

File: model_analyzer/record/metrics_manager.py [hunk 6]
```diff
@@ -367,6 +367,7 @@ def _create_model_variant(
                         self._output_model_repo_path, variant_name
                     )
             except FileExistsError:
+                # Ignore if the file already exists
```
Comment (nv-hwoo) [inline model_analyzer/record/metrics_manager.py:370]: Unnecessary comment?

**Review by nv-hwoo — APPROVED** [review]
Just a few nitpickings and all LGTM.

File: model_analyzer/perf_analyzer/perf_config.py (hunk 4 above)
Comment (pskiran1) [reply model_analyzer/perf_analyzer/perf_config.py:298]: This comment is intended to resolve https://github.com/triton-inference-server/model_analyzer/security/code-scanning/8

File: model_analyzer/perf_analyzer/perf_analyzer.py (hunk 5 above)
Comment (pskiran1) [reply model_analyzer/perf_analyzer/perf_analyzer.py:352]: This comment is intended to resolve https://github.com/triton-inference-server/model_analyzer/security/code-scanning/7

File: model_analyzer/record/metrics_manager.py (hunk 6 above)
Comment (pskiran1) [reply model_analyzer/record/metrics_manager.py:370]: This comment is intended to resolve https://github.com/triton-inference-server/model_analyzer/security/code-scanning/6

**Pull request merged**